src/agent.cpp

```diff
@@ -778,6 +793,242 @@ Agent::doMove()
 
 /*-------------------------------------------------------------------*/
 /*!
+ * This Action marks the player with the specified uniform number.
+*/
+bool Agent::doMarkPlayer(int unum) {
+  const WorldModel & wm = this->world();
+  Vector2D kicker_pos = Vector2D :: INVALIDATED;
+  Vector2D player_pos =  Vector2D :: INVALIDATED;
+  int kicker_unum = -1;
+  const PlayerPtrCont::const_iterator o_end = wm.opponentsFromSelf().end();
+  int count = 0;
+  for ( PlayerPtrCont::const_iterator it = wm.opponentsFromSelf().begin(); it != o_end; ++it ) {
+      if ( (*it)->distFromBall() < 5 ) {
+          kicker_pos = (*it)->pos();
+          kicker_unum = (*it)->unum();
+      }
+  }
+
+  for ( PlayerPtrCont::const_iterator it = wm.opponentsFromSelf().begin(); it !=  o_end; ++it ) {
+	  if ( (*it)-> unum() == unum ) {
+           player_pos = (*it)->pos();
+      }
+  }
+
+  if (!player_pos.isValid()) {
+      //Player to be marked not found
+      return false;
+  }
+  if (!kicker_pos.isValid()) {
+      //Kicker not found
+      return false;
+  }
+  if (unum == kicker_unum || kicker_pos.equals(player_pos)) {
+    //Player to be marked is kicker
+    return false;
+  }
+  double x = player_pos.x + (kicker_pos.x - player_pos.x)*0.1;
+  double y = player_pos.y + (kicker_pos.y - player_pos.y)*0.1;
+  Body_GoToPoint(Vector2D(x,y), 0.25, ServerParam::i().maxDashPower()).execute(this);
+  return true;
+}
+
+/*!
+ * This Action marks the player which is "nearindex" nearest to the ball.
```

Comment:
nearIndex=1 would mark the player 1st  closest to the ball.

It  is an unused function. However maybe useful in case we want to replace the current version of mark player to account for nearness instead of just the uniform numbers.

Removing this function for the time being